llm/tests/test_cli.py

```
@@ -96,7 +96,51 @@ def test_load_serving_config_defaults_npu_page_size(tmp_path):
+
+def test_load_serving_config_rejects_l3_max_batch_size_mismatch(tmp_path):
+    model_dir = tmp_path / "model"
+    model_dir.mkdir()
+    config_data = _base_config(model_dir, backend="npu")
+    config_data["runtime"]["max_batch_size"] = 1
+    config_data["npu"] = {
+        "l3": True,
+    }
+    config_path = _write_config(tmp_path, config_data)
+
+    with pytest.raises(ValueError, match="runtime.max_batch_size=16"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)